src/Mod/Surface/App/FeatureExtend.cpp

```diff
@@ -46,16 +46,61 @@ const App::PropertyFloatConstraint::Constraints ToleranceRange = {0.0,10.0,0.01}
 const App::PropertyFloatConstraint::Constraints ExtendRange = {-0.5,10.0,0.01};
 PROPERTY_SOURCE(Surface::Extend, Part::Spline)
 
+void Surface::Extend::onChanged(const App::Property* prop)
+{
+  // using a mutex and lock to protect a recursive calling when setting the new values
+  if (!lockOnChangeMutex.try_lock()) return;
+  lockOnChangeMutex.unlock();
+  std::lock_guard<std::mutex> lock(lockOnChangeMutex);
+
+  if ( ExtendUSymetric.getValue() )
+  {
+    if (prop->getName() == ExtendUNeg.getName()
```

Comment:
Please use 4-space indentation. Using 2 spaces is too little. The rest of the C++ code has 4 space indentation so you should use the same style.

Reply:
thanks for the hint and sorry for the wrong formatting.
Increased the number of spaces to 4.